src/libexpr/primops.cc

```diff
@@ -4494,9 +4494,9 @@ void EvalState::createBaseEnv()
     addConstant("__nixPath", v, {
         .type = nList,
         .doc = R"(
-          List of search path entries used to resolve [lookup paths](@docroot@/language/constructs/lookup-path.md).
+          The value of the [`nix-path` configuration setting](@docroot@/command-ref/conf-file.md#conf-nix-path), which can be overridden with the [`NIX_PATH` environment variable](@docroot@/command-ref/env-common.md#env-NIX_PATH) or the [`-I` command line option](@docroot@/command-ref/opt-common.md#opt-I).
```

Comment:
Actually I am not so sure about this change. We're replacing language that is implementation agnostic ("list of search path entries used to resolve lookup paths") with something that is implementation specific (blah blah blah nasty global configuration setting). I think that is not good.

Better to still introduce it in the implementation-agnostic way, and then say that the value comes from this setting.

Reply:
I thought so too after revisiting it, but realized that it's probably not smart to say what something is *intended for* rather than what something *is*. And in this case it *is* the value of that global setting.

I don't see the value of being implementation-agnostic at this exact point. Most built-ins are very deeply entangled in "historical reasons", and that's okay. Can't change it with talking about it differently.

Reply:
But

> List of search path entries used to resolve lookup paths

is not some mere opinion; it's a true fact.

Remember, all I am saying is keep that as the opening sentence, and then the new bit can go somewhere else.

Reply:
Indeed I think the plan was to add something like the new bit all along, but after the NIX_PATH vs settings precedence issues were figured out, which ended up never happening.